components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -146,41 +146,36 @@ mod parser {
         ctx: &mut EvalContext,
         input: &'a str,
         fsp: u8,
-    ) -> IResult<&'a str, [u32; 4], ()> {
+    ) -> IResult<&'a str, Duration, ()> {
         let (rest, digits) = digit1(input)?;
         if digits.len() == 12 || digits.len() == 14 {
-            let datetime = DateTime::parse_datetime(ctx, digits, fsp as i8, true)
+            let datetime = DateTime::parse_datetime(ctx, input, fsp as i8, true)
                 .map_err(|_| nom::Err::Error(()))?;
-            let (rest, fraction) = fraction(input, fsp)?;
             return Ok((
-                rest,
-                [
-                    datetime.hour(),
-                    datetime.minute(),
-                    datetime.second(),
-                    fraction,
-                ],
+                &input[input.len()..],
```

Comment:
Do we need to `input[input.len()..]` because the rest is always `""`;

Reply:
OK. Seems the whole function doesn't need to return an `IResult`, a simple `Result` is enough.